feat: add input-otp

shadcn opened the pull request:
Based on https://input-otp.rodz.dev

shadcn commented:
@damianricobelli Agreed. I've added examples for custom patterns/regex and separator. I'll add controlled, form and server actions.

damianricobelli commented:
> @damianricobelli Agreed. I've added examples for custom patterns/regex and separator. I'll add controlled, form and server actions.

Sounds good. If you need help with this let me know

guilhermerodz commented:
hi @shadcn there's a minor update on `input-otp`. It doesn't change any logic on the input itself, just added a CSS rule for iOS.

before you release it, feel free to update to `1.0.1`.

shadcn commented:
@guilhermerodz the cli will pick the latest version when you add the component.

guilhermerodz reviewed (approved):
looks good :) I'd just update the input version for the demo

shadcn commented:
@damianricobelli I've added the `controlled` and `form` examples. Server actions will come in another update.

@guilhermerodz I've updated to the latest version.

brunoeduardodev commented on apps/www/content/docs/components/input-otp.mdx:
```diff
@@ -0,0 +1,202 @@
+  maxLength={6}
+  render={({ slots }) => (
+    <InputOTPGroup className="gap-2">
+      {slots.map((slot, index) => (
+        <>
+          <InputOTPSlot key={index} className="rounded-md border" {...slot} />
+          {index !== slots.length - 1 && <InputOTPSeparator />}
+        </>
+      ))}{" "}
+    </InputOTPGroup>
```
isn't this example actually not ideal since the react fragment is not receiving a key?

shouldn't this be replaced to, for example

```tsx
<React.Fragment key={index}>
    <InputOTPSlot className="rounded-md border" {...slot} />
    {index !== slots.length - 1 && <InputOTPSeparator />}
</React.Fragment>
```

shadcn replied:
Fixed. Thank you.

guilhermerodz reviewed (commented):
hi @shadcn I'm not sure if I'm seeing :disabled modifiers, let me check this out

guilhermerodz commented on apps/www/registry/default/ui/input-otp.tsx:
```diff
@@ -0,0 +1,64 @@
+>(({ className, ...props }, ref) => (
+  <OTPInput
+    ref={ref}
+    containerClassName={cn("flex items-center gap-2", className)}
```
Add `has-[:disabled]:opacity-30` for disabled modifier appearance.
PS: do the same for `registry/default` and `registry/new-york`

@shadcn 

```diff
-containerClassName={cn("flex items-center gap-2", className)}
+containerClassName={cn("flex items-center gap-2 has-[:disabled]:opacity-30", className)}
```

guilhermerodz replied:
@shadcn I have added the modifier for you in this PR. https://github.com/shadcn-ui/ui/pull/2938

guilhermerodz reviewed (changes requested):
@shadcn Looks good. Just need to add the `has-[:disabled]` modifier to the container.

Sn0wye commented:
Hey @shadcn, currently the Input OTP is missing on the website navbar & command bar, but via url it's working fine
